mne/preprocessing/annotate_nan.py

```diff
@@ -0,0 +1,63 @@
+# Author: David Julien <david.julien@ifsttar.fr>
+#
+# License: BSD (3-clause)
+
+from ..annotations import (Annotations)
+
+
+def annotate_nan(raw):
+    """Detect segments with NaN and annotate them.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Data to find segments with NaN values.
+
+    Returns
+    -------
+    annot : instance of mne.Annotations
+        Periods with NaN segments annotated as bad_NAN.
+    """
+    annot = raw.annotations
+
+    annotations_dict = {}
+    sampling_duration = 1 / float(raw.info['sfreq'])
+    data, times = raw.get_data(return_times=True)
+    for row, channel_values in enumerate(data):
+        for time, value in zip(times, channel_values):
+            if str(value) == 'nan':
+                print("nan @ %d" % time)
+                annotations_dict[time] = sampling_duration
+
+    tolerance = 0.02  # we're comparing floats, so rounding might be a problem
+    to_delete = []  # one cannot change the size of a dict during iteration
+
+    for old_key in annotations_dict.keys():
+        for new_key in annotations_dict.keys():
+            old_value = annotations_dict[old_key]
+            if old_key < new_key <= (float(old_key) + old_value + tolerance):
+                # the data is processed from start to end,
+                # so keys are already sorted
+                if old_key + old_value <= new_key + annotations_dict[new_key]:
+                    diff = float(new_key) + annotations_dict[new_key] - \
+                           (old_key + old_value)
+                    annotations_dict[old_key] += diff
+                    to_delete.append(new_key)
```

Comment:
It seems rather dirty to me, but I couldn't come with a more simple way to merge segments. Any ideas ?